src/sentry/lang/javascript/cache.py

```diff
@@ -39,19 +37,17 @@ def alias(self, alias, target):
     def add(self, url, source, encoding=None):
         url = self._get_canonical_url(url)
 
-        if not isinstance(source, SourceView):
-            if isinstance(source, str):
-                source = source.encode("utf-8")
-            # If an encoding is provided and it's not utf-8 compatible
-            # we try to re-encoding the source and create a source view
-            # from it.
-            elif encoding is not None and not is_utf8(encoding):
-                try:
-                    source = source.decode(encoding).encode("utf-8")
-                except UnicodeError:
-                    pass
-            source = SourceView.from_bytes(source)
-        self._cache[url] = source
+        if isinstance(source, str):
+            source = source.encode("utf-8")
+        # If an encoding is provided and it's not utf-8 compatible
+        # we try to re-encode the source.
+        elif encoding is not None and not is_utf8(encoding):
+            try:
+                source = source.decode(encoding).encode("utf-8")
+            except UnicodeError:
+                pass
+
+        self._cache[url] = source.decode("utf-8")
```

Comment:
we are now doing an `encode` (turn `str` into `bytes`) / `decode` (turn `bytes` int `str`) unconditionally which I think we can avoid.

I think the intention previously was to convert to `bytes` as input to `SourceView`. But now we want the cache to be `str`, so no need for this roundtrip.